controller/events.go

```diff
@@ -170,9 +170,8 @@ func (e *EventListener) Listen() error {
 // Notify notifies all sbscribers of the given event.
 func (e *EventListener) Notify(event *ct.AppEvent) {
 	e.subMtx.RLock()
-	subscribers := e.subscribers
-	e.subMtx.RUnlock()
-	if subs, ok := subscribers[event.AppID]; ok {
+	defer e.subMtx.RUnlock()
```

Comment:
Locking the mutex for the duration of `sub.Notify` can cause issues because it can call [sub.CloseWithError](https://github.com/flynn/flynn/blob/master/controller/events.go#L49), which ends up calling [l.Unsubscribe](https://github.com/flynn/flynn/blob/master/controller/events.go#L68), which then [tries to lock the mutex](https://github.com/flynn/flynn/blob/master/controller/events.go#L123).